src/app/components/dialogs/dialog.service.ts

```diff
@@ -31,4 +33,8 @@ export class DialogService extends BaseDialogService {
   public openScenarioTypeDialog(): Promise<DialogData> {
     return this.openDialog('scenarioTypeDialog', ScenarioTypeDialogComponent);
   }
+
+  public openMapSettingsDialog(colourGradient: ColourGradientType): Promise<DialogData> {
```

Comment:
Could have some types on the dialogData to make it easier to work out what's being passed in and out.

Reply:
Dialog data is an already defined data Type, you're required to pass in type sensitive data? Do you mean the dataIn and dataOut data types?